lib/hooks/index.js

```diff
@@ -116,6 +122,16 @@ function activate(agent) {
         if (instrumentation && agent.config().excludedHooks &&
             agent.config().excludedHooks.indexOf(request) === -1 &&
             !instrumentation.hook) { // not already patched.
+          var mainScriptDir = path.dirname(require.resolve(request));
+          var resolvedModule = Module._resolveLookupPaths.apply(this, arguments);
+          var paths = resolvedModule[1];
+          for (var i = 0, PL = paths.length; i < PL; i++) {
+            if (mainScriptDir.indexOf(paths[i]) === 0) {
+              var pjson = originalModuleLoad(path.join(paths[i], request, 'package.json'));
```

Comment:
This might throw. Not sure if we are 100% confidence that it can't. I am still digesting the code, so I might get to 100%.